language/tools/move-package/src/resolution/lock_file/schema.rs

```diff
@@ -58,6 +58,9 @@ pub struct Dependency {
 
     /// Expected hash for the source and manifest of the package being depended upon.
     pub digest: Option<String>,
+
+    /// Optional dependency override to handle dependency conflicts.
+    pub dep_override: Option<bool>,
```

Comment:
I don't think dep overrides should be necessary in the lock file -- i.e. the lock file should only contain one package for each package name.

Reply:
The thought here (though not fully implemented yet) was that there may be an override encoded in the dependency graph coming from an external resolver. In particular, for the following manifest for a package involving a diamond problem down the line (e.g., https://github.com/move-language/move/pull/964/files#diff-8c1d4a846826c7a3c78ecd708834e6d958c922a948298f9279e402ab3b30ea37) , instead of overriding `B` an external resolver could conceivably contain an override for `A`.

If we don't worry about this case (I think the external override would only make sense at the level of external root package) then we indeed don't need overrides in Move.lock files.

Reply:
Let's not worry about external resolvers specifying overrides (for now).